src/screens/CreatePost/index.js

```diff
@@ -506,14 +509,23 @@ const CreatePost = () => {
         });
       }
 
-      await createPost(data);
+      const post = await createPost(data);
       if (params.onRefresh && typeof params.onRefresh === 'function') {
         params.onRefresh();
       }
-      showMessage({
-        message: StringConstant.createPostDone,
-        type: 'success'
-      });
+      if (post.code === 200) {
+        showMessage({
+          message: StringConstant.createPostDone,
+          type: 'success'
+        });
+        setLoadingPost(false);
+      } else {
+        showMessage({
+          message: StringConstant.createPostFailedGeneralError,
+          type: 'danger'
+        });
+        setLoadingPost(false);
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The handling of the post response is well done here. The loading state is correctly reset after the post operation regardless of whether it was successful or not. However, consider adding error handling for the case when `createPost(data)` throws an exception. In the current implementation, if `createPost(data)` throws an exception, `setLoadingPost(false)` won't be called, which might leave your application in a perpetual loading state.
```diff
      const post = await createPost(data);
       if (params.onRefresh && typeof params.onRefresh === 'function') {
         params.onRefresh();
       }
       if (post.code === 200) {
         showMessage({
           message: StringConstant.createPostDone,
           type: 'success'
         });
         setLoadingPost(false);
       } else {
         showMessage({
           message: StringConstant.createPostFailedGeneralError,
           type: 'danger'
         });
         setLoadingPost(false);
       }
+    } catch (e) {
+      setLoadingPost(false);
       if (__DEV__) {
         console.log('CreatePost : ', e);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->